### PR TITLE
feat(api): introduce a way to get viewers of a bossbar

### PR DESCRIPTION
**Pull request by kashike**

### CODE REVIEW

File: api/src/main/java/net/kyori/adventure/bossbar/BossBarViewer.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * SOFTWARE.
+ */
+package net.kyori.adventure.bossbar;
+
+/**
+ * Something that can view a {@link BossBar}.
+ *
+ * @since 4.14.0
+ */
+public interface BossBarViewer {
```
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/bossbar/BossBarViewer.java:34]: Perhaps methods for checking if a viewer can see a boss bar and for getting the boss bars they can see?

**Review by kezz — CHANGES_REQUESTED** [review]

File: api/src/main/java/net/kyori/adventure/bossbar/BossBar.java [hunk 2]
```diff
@@ -379,6 +380,42 @@ default float percent() {
...
+   *
+   * <p>The returned value may be empty if this method is unsupported.</p>
+   *
+   * @return an unmodifiable view of the viewers of this bossbar
+   * @since 4.14.0
+   */
+  @NotNull @UnmodifiableView Iterable<? extends BossBarViewer> viewers();
+
+  /**
+   * Show this bosbar to {@code viewer}.
```
Comment (KingOfSquares) [inline api/src/main/java/net/kyori/adventure/bossbar/BossBar.java]:
```suggestion
   * Show this bossbar to {@code viewer}.
```

File: api/src/main/java/net/kyori/adventure/bossbar/BossBar.java [hunk 3]
```diff
@@ -379,6 +380,42 @@ default float percent() {
...
+   * @see Audience#showBossBar(BossBar)
+   * @since 4.14.0
+   */
+  default @NotNull BossBar addViewer(final @NotNull Audience viewer) {
+    viewer.showBossBar(this);
+    return this;
+  }
+
+  /**
+   * Hide this bosbar from {@code viewer}.
```
Comment (KingOfSquares) [inline api/src/main/java/net/kyori/adventure/bossbar/BossBar.java]:
```suggestion
   * Hide this bossbar from {@code viewer}.
```

**Review by KingOfSquares — CHANGES_REQUESTED** [review]

**Review by KingOfSquares — APPROVED** [review]

File: api/src/main/java/net/kyori/adventure/bossbar/BossBarViewer.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  /**
+   * Gets an unmodifiable view of all bossbars currently being viewed.
+   *
+   * @return an unmodifiable view of all bossbars currently being viewed
+   * @since 4.14.0
+   */
+  @UnmodifiableView
+  @NotNull Iterable<? extends BossBar> viewedBossBars();
```
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/bossbar/BossBarViewer.java]: Don't like this method name - sounds past tense. Perhaps just `bossBars`

**Review by kezz — CHANGES_REQUESTED** [review]

File: api/src/main/java/net/kyori/adventure/bossbar/BossBarViewer.java (hunk 4 above)
Comment (MiniDigger) [reply api/src/main/java/net/kyori/adventure/bossbar/BossBarViewer.java]: activeBossBars maybe?

**Review by kezz — APPROVED** [review]

**Review by zml2008 — APPROVED** [review]

**Pull request merged**